File: regexlib/2021-5-15/javascript_test_file/regexlib_8048.js
/* 8048
 * ^(http(s?):\/\/)(www.)?(\w|-)+(\.(\w|-)+)*((\.[a-zA-Z]{2,3})|\.(aero|coop|info|museum|name))+(\/)?$
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"http://1"+".aero.AA"*1024+"! _1_EOD(i2)"
 */
var REGEX = new RegExp("^(http(s?):\/\/)(www.)?(\w|-)+(\.(\w|-)+)*((\.[a-zA-Z]{2,3})|\.(aero|coop|info|museum|name))+(\/)?$")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = 'http://1' + '.aero.AA'.repeat(i*1) + '! _1_EOD(i2)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/python_re2_test_file/regexlib_8048.py
# 8048
# ^(http(s?):\/\/)(www.)?(\w|-)+(\.(\w|-)+)*((\.[a-zA-Z]{2,3})|\.(aero|coop|info|museum|name))+(\/)?$
# EXPONENT
# nums:5
# EXPONENT AttackString:"http://1"+".aero.AA"*1024+"! _1_EOD(i2)"

import re2 as re
from time import perf_counter

regex = """^(http(s?):\/\/)(www.)?(\w|-)+(\.(\w|-)+)*((\.[a-zA-Z]{2,3})|\.(aero|coop|info|museum|name))+(\/)?$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "http://1" + ".aero.AA" * i * 1 + "! _1_EOD(i2)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")